credoai/evaluators/shap.py

```diff
@@ -131,6 +136,25 @@ def _get_mult_sample_shapley_values(self) -> DataFrame:
         res.name = "Shap values for specific samples"
         return res
 
+    def _validate_samples_ind(self, limit=5):
+        """
+        Enforce limit on maximum amount of samples for which to extract
+        individual shap values.
+
+        Parameters
+        ----------
+        limit : int, optional
+            Max number of samples allowed, by default 5
+
+        Raises
+        ------
+        ValidationError
+        """
+        if self.samples_ind is not None:
+            if len(self.samples_ind) > limit:
+                message = "The maximum amount of individual samples_ind allowed is 5."
```

Comment:
Maybe would be good to explain that we're doing this to avoid leaking too much PII. Theoretically someone could just iteratively call the sample function with different subsets of indices; not much we can do about that. It's at least good to highlight the "why?" so it doesn't seem arbitrary.

Reply:
It doesn't look like there are test cases for SHAP. I'm curious what that might look like?

Reply:
Probably a general run test, parametrized on sampling strategy.